### DISCUSSION TITLE
Planet zodiac

### DISCUSSION THREAD

**Discussion by NexLeapIO** [cosinekitty/astronomy#338] [Q&A] [answered]
Can someone help me with a few questions?
1. Explain how I can calculate which zodiac sign a particular planet, Sun or Moon is currently in depending on the position of the observer?
2. How can I calculate what is the age/day of the moon at the moment?

**Comment by cosinekitty** [comment]
Astronomy Engine supports multiple programming languages. What programming language are you using? There are demos for calculating planet positions. Once you have equatorial coordinates of a planet, you can call the "Constellation" function to determine what constellation the planet is inside.

**Reply by NexLeapIO** [reply]
Node JS. Found my self how to find zodiac sign. Also made function my self for age.. But maybe there is build one already?

**Reply by NexLeapIO** [reply]
Also I am interested how can I find moon phase angle from which it starts. I mean we all from different coordinates see it different. I am creating some application which shows current moon phase based on location, so I wanna rotate it by this degree too. Is it possible?

**Reply by matheo** [reply]
I wonder about the precision of your results.
Did they match the sign+degrees of other softwares out there?

**Reply by NexLeapIO** [reply]
I mean I used ur function for zodiac. For moon age I just find next new moon after 2000 janurary 1 and calculated it. What about angle of phase? any suggestions?

**Reply by cosinekitty** [reply]
Most calendars use the Moon's geocentric ecliptic longitude to determine the phase. New Moon occurs when the Moon's ecliptic longitude is the same as the Sun's. First quarter occurs when the longitude difference is 90&deg;, etc. There is a function [MoonPhase](https://github.com/cosinekitty/astronomy/tree/master/source/js#MoonPhase) that is provided just for this purpose. You can also use [SearchMoonPhase](https://github.com/cosinekitty/astronomy/tree/master/source/js#searchmoonphasetargetlon-datestart-limitdays--astrotime--null) to find the next time the Moon reaches a desired phase angle.

**Reply by NexLeapIO** [reply]
I mean I want get rotation from observer too, like this 
![image](https://github.com/cosinekitty/astronomy/assets/164171686/824f064b-0aba-4396-9912-36a2343d2b45)

**Reply by cosinekitty** [reply, marked as answer]
You should take a look through the Node.js demos, and particularly at the [Camera](https://github.com/cosinekitty/astronomy/tree/master/demo/nodejs#camera) demo.
